### PR TITLE
[ClangImporter] EnumInfo improvements

### PR DESCRIPTION
**Pull request by jrose-apple**
- Handle `ns_error_domain` on anonymous enums with typedefs (instead of crashing)
- Stop copying EnumInfo strings into Swift's ASTContext (they're kept alive by Clang's ASTContext already)

### CODE REVIEW

**Comment by jrose-apple** [comment]
@swift-ci Please test

File: lib/ClangImporter/ImportEnumInfo.cpp [hunk 1]
```diff
@@ -308,6 +307,8 @@ void EnumInfo::determineConstantNamePrefix(ASTContext &ctx,
     // Don't use importFullName() here, we want to ignore the swift_name
     // and swift_private attributes.
     StringRef enumNameStr = decl->getName();
+    if (enumNameStr.empty())
+      enumNameStr = decl->getTypedefNameForAnonDecl()->getName();
```
Comment (milseman) [inline lib/ClangImporter/ImportEnumInfo.cpp]: Assert / error that it now has a name?

**Review by milseman — APPROVED** [review]
LGTM

**Review by milseman — APPROVED** [review]
🎉

File: lib/ClangImporter/ImportEnumInfo.cpp (hunk 1 above)
Comment (jrose-apple) [reply lib/ClangImporter/ImportEnumInfo.cpp]: Ah, yeah, that's how it got caught originally, but I guess it'd be nicer to catch it here than in `getCommonPluralPrefix`.

**Comment by jrose-apple** [comment]
@swift-ci Please smoke test

**Pull request merged**